### PR TITLE
Reset password

### PR DESCRIPTION
**Pull request by leventdem**

### CODE REVIEW

File: src/library/masq.js [hunk 1]
```diff
@@ -271,6 +270,24 @@ class Masq {
...
+   */
+  async updatePassphrase (currentPassPhrase, newPassPhrase) {
+    this._checkProfile()
+    const protectedMK = await this._getProtectedMK()
+    try {
+      const protectedMKNewPass = await updateMasterKeyAndNonce(currentPassPhrase, newPassPhrase, protectedMK)
+      // we do not use this.encryptAndPut because this value is not encrypted
+      await this.profileDB.putAsync('/profile/protectedMK', protectedMKNewPass)
+    } catch (error) {
+      this.closeProfile()
```
Comment (G-Ray) [inline src/library/masq.js]: I think we should not close the profile, only throw. This way we will be able to display an error and the user will be able to retry to update his pass

File: test/masq.test.js [hunk 2]
```diff
@@ -137,6 +138,15 @@ describe('masq internal operations', function () {
     expect(encryptedMasterKeyAndNonce.ciphertext).to.exist
   })
 
+  it('should update protectedMK after a passphrase change(must be logged)', async () => {
+    const protectedMasterKey = await masq._getProtectedMK()
+    const { encryptedMasterKeyAndNonce } = protectedMasterKey
+
+    await masq.updatePassphrase(PASSPHRASE, `${PASSPHRASE}new`)
+    const protectedMasterKeyNewPass = await masq._getProtectedMK()
+    expect(encryptedMasterKeyAndNonce.ciphertext).to.not.equal(protectedMasterKeyNewPass.encryptedMasterKeyAndNonce.ciphertext)
```
Comment (G-Ray) [inline test/masq.test.js]: wow can you reduce the length of this line please ?

Comment (G-Ray) [reply test/masq.test.js]: by destructuring the object for instance

File: src/library/masq.js (hunk 1 above)
Comment (sujeebant) [reply src/library/masq.js]: :+1:

**Review by sujeebant — APPROVED** [review]

**Pull request merged**